Number_types/include/CGAL/FPU.h

```diff
@@ -347,16 +346,32 @@ inline double IA_bug_sqrt(double d)
 // With GCC, we can do slightly better : test with __builtin_constant_p()
 // that both arguments are constant before stopping one of them.
 // Use inline functions instead ?
-#define CGAL_IA_ADD(a,b) CGAL_IA_FORCE_TO_DOUBLE((a)+CGAL_IA_STOP_CPROP(b))
-#define CGAL_IA_SUB(a,b) CGAL_IA_FORCE_TO_DOUBLE(CGAL_IA_STOP_CPROP(a)-(b))
-#define CGAL_IA_MUL(a,b) CGAL_IA_FORCE_TO_DOUBLE(CGAL_IA_STOP_CPROP(a)*CGAL_IA_STOP_CPROP(b))
```

Comment:
You dropped CGAL_IA_FORCE_TO_DOUBLE for the normal case, that looks wrong.

Reply:
Yes, I think you're right -- I didn't look deep enough into all of the cases it handles.

How about putting that into the other branch of CGAL_IA_UP?

Or is that getting too overloaded?

Reply:
Putting it in the other branch of CGAL_IA_UP sounds fine.

Reply:
Done